examples/fedopt_example/metrics.py

```diff
@@ -75,20 +75,26 @@ def compute_metrics(self) -> Metrics:
         return metrics
 
 
-class CustomMetricMeter(MetricMeter):
-    def __init__(self, label_encoder: LabelEncoder) -> None:
+class CompoundMetric(Metric):
+    def __init__(self, name: str) -> None:
         """
         This class is used to compute metrics associated with the AG's News task. There are a number of classes and
         we want to accumulate a bunch of statistics all at once to facilitate the computation of a number of different
-        metrics for this problem. As such, we define our own MetricMeter and bypass the standard metric meter
-        implementations, which calculate separate metrics individually.
+        metrics for this problem. As such, we define our own Metric class and bypass the standard SimpleMetric class,
+        which calculate separate metrics individually.
 
         Args:
-            label_encoder (LabelEncoder): This class is used to determine the mapping of integers to label names for
-            the AG's news task.
+            name (str): The name of the compound metric.
         """
+        super().__init__(name)
         self.true_preds = 0
         self.total_preds = 0
+        self.classes: List[str]
+        self.label_to_class: Dict[int, str]
+        self.n_classes: int
+        self.outcome_dict: Dict[str, Outcome]
+
+    def _setup(self, label_encoder: LabelEncoder) -> None:
```

Comment:
Typically the `_` prefix is reserved for protected or private methods. That is, methods that are exclusively called within the class itself, rather than externally. All that is to say, I would recommend dropping the _ based on the way this is being used 🙂

Reply:
Yeah, good call! I initially was aiming for it to be internal but realized we have to call it externally so I forgot to change it back